[RNMobile] Improvements to Getting Started Guides

SiobhyB opened the pull request:
## What?

With this PR, some updates are made to the Getting Started guides for the React Native demo app. 

## Why?

These updates were prompted by some recent recommendations for M1 machines (including a JDK recommendation). By updating the MacOS set up guide and linking to React Native's doc in the general guide, it is hoped the docs will make the set up process smoother for newcomers.    

## How?

### Getting Started - General Guide

The following paragraph has been added to the [general guide](https://github.com/WordPress/gutenberg/blob/44ed96f337400f90c2a193981810bd6bf0bc7f3f/docs/contributors/code/react-native/getting-started-react-native.md):

```
If the Android emulator doesn't start correctly, or compiling fails with `Could not initialize class org.codehaus.groovy.runtime.InvokerHelper` or similar, it may help to double check the set up of your development environment against the latest requirements in [React Native's documentation](https://reactnative.dev/docs/environment-setup). With Android Studio, for example, you will need to configure the `ANDROID_HOME` environment variable and ensure that your version of JDK matches the latest requirements.
```
This is taken from the guide in [the Gutenberg Mobile repository](https://github.com/wordpress-mobile/gutenberg-mobile/blob/3f2cc98b29159d66917dc2cf13e1416d487f987f/README.md) and brings the Gutenberg guide in alignment. An aim behind linking directly to React Native's official documentation is to guide anyone who may be struggling with issues related to their JDK version or similar, as we can trust the official documentation to have the most recent recommendations.

### Getting Started - In-Depth MacOS Guide

The following is an overview of the main changes that have been made to the [more in-depth MacOS guide](https://github.com/WordPress/gutenberg/blob/44ed96f337400f90c2a193981810bd6bf0bc7f3f/docs/contributors/code/react-native/osx-setup-guide.md):

* The sections have been rearranged in order to improve the overall flow of the guide. There are now five main sections, arranged as follows: 1. Clone Gutenberg. 2. iOS. 3. Android. 4. Unit Tests. 5. Integration Tests. 
* A [Cocoapods sub-section](https://github.com/WordPress/gutenberg/blob/44ed96f337400f90c2a193981810bd6bf0bc7f3f/docs/contributors/code/react-native/osx-setup-guide.md?plain=1#L45-L58) has been added under the iOS section.
* A [Java Development Kit (JDK) sub-section](https://github.com/WordPress/gutenberg/blob/44ed96f337400f90c2a193981810bd6bf0bc7f3f/docs/contributors/code/react-native/osx-setup-guide.md?plain=1#L101-L110) has been added under the Android section. 
* The images have been set to the same width in order to tidy up the overall look of the document. 
* Several tweaks to the copy have been made, with an aim to keep the tone concise and inline with [the copy guidelines](https://github.com/WordPress/gutenberg/blob/089acde6d86bf4c792427932578bb8a5554dcdca/docs/contributors/documentation/copy-guide.md).

## Testing Instructions

* Visit the guides directly (vs. looking at the "files changed", as the formatting is hard to read). 
* Review for overall accuracy, clarity, and completeness.

SiobhyB committed:
Update with copy from Gutenberg Mobile

Gutenberg Mobile's readme had some copy in the troubleshooting section that wasn't included in Gutenberg's how-to: https://github.com/wordpress-mobile/gutenberg-mobile/edit/trunk/README.md

This commit fixes that issue by updating Gutenberg's how-to with the latest from Gutenberg Mobile.

SiobhyB committed:
Tweak wording, update inconsistent capitalisation

SiobhyB committed:
Tweaks to copy and formatting

With this commit, I've tidied up some of the copy, attempting to make the instructions as concise and helpful as possible. I've also set the images to the same width to tidy up the overall look of the document (the images will appear small on mobile devices, but can be tapped on to get the full image).

SiobhyB committed:
Set all images to fixed width, for consistency

SiobhyB committed:
Rearrange sections to improve overall flow

- The "Unit Tests" is moved towards the bottom of the document, so that's it's closer to the instructions for integration tests. the document flows as follows:
- The steps for installing Xcode are moved under the "iOS" section.
- The overall flow of the document has been updated to the following: Clone Gutenberg > iOS > Android > Tests

SiobhyB committed:
Add details about recommended JDK

SiobhyB committed:
Add details about Cocoapods

hypest commented on docs/contributors/code/react-native/osx-setup-guide.md:
````diff
@@ -71,28 +40,46 @@ npm run distclean
+[CocoaPods](https://guides.cocoapods.org/using/getting-started.html) is needed to fetch React and third-party dependencies:
+
+```
+sudo gem install cocoapods
````
I wonder if it makes sense to add some mention that this assumes using the system Ruby (hence the need for `sudo`). Additionally, though I'm not sure this has to be done in this PR, perhaps we can start recommending the Ruby version manager we mainly use (`rbenv`) as part of the setup. I think in that case, one doesn't need to manually install the `cocoapods` gem. Thoughts?

SiobhyB replied:
That's a good idea! I've gone ahead to make an update in https://github.com/WordPress/gutenberg/pull/40964/commits/421a1a802c6906ac26bed785e74dfba7ca2e3c8b. I kept the mention of how to install the gem in the case that someone is only using the system default. My thought behind that is that it may be simpler if the person only wants to use the demo app (and doesn't need to set up the native app). What do you think?

hypest replied:
makes sense to me 👍

hypest commented on docs/contributors/code/react-native/osx-setup-guide.md:
````diff
@@ -71,28 +40,46 @@ npm run distclean
-[react native doctor](https://reactnative.dev/blog/2019/11/18/react-native-doctor). From your checkout, or
-relative to `/packages/react-native-editor folder`, run:
+```
+sudo arch -x86_64 gem install ffi
````
Similar to my [comment above](https://github.com/WordPress/gutenberg/pull/40964/files#r872416814), my impression is that one doesn't need to manually install or set the architecture for the `ffi` gem when using `rbenv` and Ruby `v2.7.4` (not `v3.x.x`).

hypest replied:
👋 @SiobhyB , noticed your [follow up review request](https://github.com/WordPress/gutenberg/pull/40964#event-6623057581), but not sure if my comment here about the `ffi` thing is addressed already 🤔. I think having to manually install a specific gen under a specific architecture can be confusing or prone to error so, I wonder if we can be better off with a different approach that avoid the manual `ffi` installation, wdyt?

SiobhyB replied:
@hypest, ah, I was hoping that [the comment here](https://github.com/WordPress/gutenberg/pull/40964#discussion_r873833823) addressed that. The comment about using `ffi` is now only under the `System Ruby` section, for people who may be using the default version of Ruby (for extra context, this was taken straight from [the React Native docs](https://reactnative.dev/docs/environment-setup)). I've now added an explicit mention that it may not need to be manually installed if using a Ruby manager like `rbenv` in https://github.com/WordPress/gutenberg/pull/40964/commits/a2dd84e9b3faec47d6bf9b4dfc093128a9911894 now. How does that seem?

hypest replied:
Ah, gotcha. Thanks for the additional commit to make it more explicit. I think it's good to go!

hypest commented on docs/contributors/code/react-native/osx-setup-guide.md:
````diff
@@ -105,37 +92,46 @@ In another terminal type:
 ```
 
-After waiting for everything to build we should see:
+After waiting for everything to build, the demo app should be running from the iOS simulator:
````
Nice one, it's better to be more explicit on what the image below is supposed to show.

hypest reviewed (commented):
Had a look @SiobhyB and these are cool updates and souping up, thanks!

I left a couple of non-trivial comments about the M1 related instructions, since I'm not sure those are needed lately. Let's discuss, thanks!

SiobhyB committed:
Refine instructions for installing Cocoapods

SiobhyB committed:
Note `ffi` may already be installed

Some versions of Ruby may not require the `ffi` to be manually installed, which is noted with this commit.

hypest reviewed (approved):
LGTM!
